### PR TITLE
Fix input redirect put

### PR DESCRIPTION
**Pull request by madarivi**
This pr should solve three issues:

1. When using the put command with redirected input in the terminal the return used for kiya execution results in an extra newline in the secret value
2. When using the put command the secret was always put into kiya, even when on conformation n was entered
3. #4 

#4 was solved by not asking for conformation when reading redirected input

### CODE REVIEW

File: utils.go [hunk 1]
```diff
@@ -39,6 +39,12 @@ func valueOrReadFrom(value string, file *os.File) string {
 	if err != nil {
 		log.Fatal("Error while reading from file", file, err)
 	}
+
+	// remove newline added to std in from command execution
```
Comment (emicklei) [inline utils.go]: how can you be sure that the newline is not part of the secret?

File: utils.go [hunk 2]
```diff
@@ -30,14 +30,11 @@ func getValueByKey(kmsService *cloudkms.Service, storageService *cloudstore.Clie
...
 }
 
-// valueOrReadFrom returns the value argument if not empty or the contents of the file argument if empty.
-func valueOrReadFrom(value string, file *os.File) string {
-	if len(value) != 0 {
-		return value
-	}
-	buffer, err := ioutil.ReadAll(file)
+// readFromStdIn tries to read tries to read required input from standard in
+func readFromStdIn() string {
```
Comment (emicklei) [inline utils.go:34]: the old function was a generalization ; why was it replaced? what if we want a future flag "-i" that explictly reads from file?  (ok we can apply generalization later)

File: cmd_move.go [hunk 3]
```diff
@@ -23,7 +23,7 @@ func commandMove(
 		log.Fatal(tre.New(err, "get source key failed", "key", sourceKey))
 	}
 	// store value for key to target
-	commandPutPasteGenerate(kmsService, storageService, target, "put", targetKey, sourceValue)
+	commandPutPasteGenerate(kmsService, storageService, target, "put", targetKey, sourceValue, true)
```
Comment (emicklei) [inline cmd_move.go:26]: tip:  create a constant with value true and use that constant here. Better readible

File: cmd_put_paste_generate.go [hunk 4]
```diff
@@ -10,12 +10,13 @@ import (
 )
 
 func commandPutPasteGenerate(kmsService *cloudkms.Service, storageService *cloudstore.Client,
-	target profile, command, key, value string) {
+	target profile, command, key, value string, prompt bool) {
```
Comment (emicklei) [inline cmd_put_paste_generate.go]: prompt -> mustPrompt

File: main.go [hunk 5]
```diff
@@ -52,13 +52,12 @@ func main() {
 
 	case "put":
 		key := flag.Arg(2)
-		var value string
 		if len(flag.Arg(3)) != 0 {
-			value = flag.Arg(3)
+			commandPutPasteGenerate(kmsService, storageService, target, "put", key, flag.Arg(3), true)
```
Comment (emicklei) [inline main.go]: better to declare variable containing flag.Arg(3).  Better readable

File: utils.go (hunk 1 above)
Comment (madarivi) [reply utils.go]: It seems to be consistent for reading from standard in, as the return from executing the command is included after the file. I'm not sure as to the why, can't find much about it. Ofcourse if we read from an actual file (say with a -file option or something) we don't want this to happen.

File: utils.go (hunk 2 above)
Comment (madarivi) [reply utils.go:34]: The generalized function was used in three places with as input os.StdIn. I removed this complexity for now.

File: main.go [hunk 6]
```diff
@@ -52,8 +57,13 @@ func main() {
 
 	case "put":
 		key := flag.Arg(2)
-		value := valueOrReadFrom(flag.Arg(3), os.Stdin)
-		commandPutPasteGenerate(kmsService, storageService, target, "put", key, value)
+		value := flag.Arg(3)
+		if len(value) != 0 {
+			commandPutPasteGenerate(kmsService, storageService, target, "put", key, value, doPrompt)
+		} else {
+			value = readFromStdIn()
```
Comment (amohabir) [inline main.go:64]: Suggestion to put the usage of stdin in the README and `--help`, if available

Comment (amohabir) [reply main.go:64]: @emicklei Just wondering about this suggestion: What about using a `-f` flag for stdin and/or file like in `kubectl`? I think that's quite elegant as a solution.

Search for stdin: https://kubernetes.io/docs/reference/kubectl/overview/

Comment (emicklei) [reply main.go:64]: +1 for the -f flag that read from file.  Would like to keep stdin as well (do not break existing usage)

**Pull request merged**